src/mersenne_twister.hpp

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019-2020 libsbn project contributors.
+// libsbn is free software under the GPLv3; see LICENSE file for details.
+
+#ifndef SRC_MERSENNE_TWISTER_HPP_
+#define SRC_MERSENNE_TWISTER_HPP_
+
+#include <random>
+
+class MersenneTwister {
+ public:
+  inline void SetSeed(unsigned long seed) { random_generator_.seed(seed); }
+  inline std::mt19937 &GetGenerator() const { return random_generator_; };
+
+ private:
+  static std::random_device random_device_;
+  static std::mt19937 random_generator_;
```

Comment:
@matsen It may be better to leave them non-static, and instantiate MerseneTwister only once with a static linkage, taking seed in the constructor

Reply:
Thanks @ognian- ! I'm not sure what you mean by "static linkage". Do you mean that we have a static MerseneTwister? I'd prefer not to do that, because I want this class to sit inside a templated class, which seemed to be causing troubles. 

We may want to set the seed after construction.

Do you see problems with this approach, or is it more just ugly?

Reply:
My point is mainly for ugliness, not that it wouldn't work. This has the appearance that the class has state, thus the users should instantiate it and use that instance. But setting the seed on any instance would set it for all instances. I think it would be cleaner if the class is a singleton.